tools/cli/src/cli/oauth2.rs

```diff
@@ -305,6 +306,25 @@ impl Oauth2Opt {
                     Err(e) => handle_client_error(e, nopt.copt.output_mode),
                 }
             }
+            Oauth2Opt::SetOrigin { nopt, origin } => {
+                let client = nopt.copt.to_client(OpType::Write).await;
+                match client
+                    .idm_oauth2_rs_update(
+                        &nopt.name,
+                        None,
+                        None,
+                        Some(origin),
+                        None,
+                        true,
```

Comment:
The bools at the end shouldn't be true, they reset all the secrets associated with the RS.

Reply:
👍 Was just a guess, didn't actually know what to do

Reply:
No worries, that's why we have a review process 😄